### DISCUSSION TITLE
Week 4

### DISCUSSION THREAD

**Discussion by Stev3eM** [SE-TINF22B6/CookHub#41] [Show and tell]
Hello backpackers,

This is our weekly discussion-blog to keep you up to date within our project progression.

For this week we have some updates for :

✅ Created activity diagrams for the current use cases
[login-activity-diagram](https://github.com/SE-TINF22B6/CookHub/blob/main/Project_Doc/Diagrams/login-activity-diagram.drawio.svg)
[logout-activity-diagram](https://github.com/SE-TINF22B6/CookHub/blob/main/Project_Doc/Diagrams/logout-activity-diagram.drawio.svg)
[create-recipe-diagram](https://github.com/SE-TINF22B6/CookHub/blob/main/Project_Doc/Diagrams/create-recipe-diagram.drawio.svg)

✅ Created two sequence diagrams for the project.
[login-sequence-diagram](https://github.com/SE-TINF22B6/CookHub/blob/main/Project_Doc/Diagrams/login-sequence-diagram.drawio.svg)
[openai-sequence-diagram](https://github.com/SE-TINF22B6/CookHub/blob/main/Project_Doc/Diagrams/openai-sequence-diagram.drawio.svg)

✅ UML overhauled
[use-case-diagram](https://github.com/SE-TINF22B6/CookHub/blob/main/Project_Doc/Diagrams/use-case-diagram.drawio.svg)

✅ Overhauled profile page mockup
<img width="300" alt="image" src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/b060bac8-b410-4a78-9c0a-72c7b17a7b9b">

✅ Updated the [srs.md](https://github.com/SE-TINF22B6/CookHub/blob/main/Project_Doc/srs.md) with activity diagrams, sequence diagrams, UML diagram and specified our use cases.

Your CookHub Team
<img width="50" alt="image" src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/cdbd0688-3eb9-46fa-a608-baf2efbaba88">

**Comment by TadaXD20** [comment]
Your weekly report is short but contains all the important information. Love that! 

One tip for the notation of sequence diagrams:
You only double the lifetime of an object, if it calls itself. Have a look at the following example : [https://developer.ibm.com/articles/the-sequence-diagram/](url)
![CookHub_Login_SequenceDiagram](https://github.com/SE-TINF22B6/CookHub/assets/75337582/4a02bec9-838a-4af3-b118-bc015b708c60)

Also your link to the SRS does not work. I had the same problem in my weekly report. I recommend to use the permalink and delete the '(url)'. At least that worked for us.

**Comment by giomskii** [comment]
This weeks report is quite concise and i think it covers all the important aspects. 👍  
I like the way you structured your Activity and Use Case diagrams as well as your srs. I think it would be better if you remove the citation marks of the written content but leave it for the pictures so it's better readable overall. I also appreciate your clean and simple ui-design for your webpage - it is very compact. Although i would have prefered another color over lime green but thats just a personal preference. 
In regards to your OpenAi sequence diagram, do you also plan to validate the user input before sending the request?
